torch/cuda/amp/grad_scaler.py

```diff
@@ -166,6 +170,9 @@ def _lazy_init_scale_growth_tracker(self, dev: torch.device) -> None:
             (), self._init_growth_tracker, dtype=torch.int32, device=dev
         )
 
+    def _is_tensor_on_supported_device(self, tensor: torch.Tensor):
+        return tensor.is_cuda or tensor.device.type == "xla"
```

Comment:
It would be good to define a wider list of what devices are accepted, and restructure this to be more like tensor device is in said_wide_list. Then, your subclass can add to the this list and pass this function. If you have other ideas to maintain the functionality of the assert in the constructor, I am down to consider those too.